src/runtime_src/core/tools/common/ReportDebugIpStatus.cpp

```diff
@@ -0,0 +1,1305 @@
+            << static_cast<double>(asmResults.StrDataBytes[i]) / 1000.0
+            << std::scientific << std::setprecision(ss)
+
+            << "  " << std::setw(16) << asmResults.StrBusyCycles[i]
+            << "  " << std::setw(16) << asmResults.StrStallCycles[i]
+            << "  " << std::setw(16) << asmResults.StrStarveCycles[i]
+            << std::endl;
+  }
+  
+}
+
+void 
+DebugIpStatusCollector::readLAPChecker(debug_ip_data* dbgIpInfo)
+{
+  // index in results is debugIpNum
+  uint64_t index = debugIpNum[LAPC];
+
+  // Get Debug Ip Name
+  std::string dbgIpName;
+  // Fill up string with 128 characters (padded with null characters)
+  dbgIpName.assign(dbgIpInfo->m_name, sizeof(dbgIpInfo->m_name));
+  // Strip away any extraneous null characters
+  dbgIpName.assign(dbgIpName.c_str());
+
+  // Get Cu and Port Name
+  std::string cuName;
+  std::string portName;
+  getCuNamePortName(dbgIpInfo->m_type, dbgIpName, cuName, portName);
+  cuNames[LAPC].emplace_back(cuName);
+  portNames[LAPC].emplace_back(portName);
+
+  size_t size = 0;
+
+  uint64_t statusRegisters[] = {
+    LAPC_OVERALL_STATUS_OFFSET,
+
+    LAPC_CUMULATIVE_STATUS_0_OFFSET, LAPC_CUMULATIVE_STATUS_1_OFFSET,
+    LAPC_CUMULATIVE_STATUS_2_OFFSET, LAPC_CUMULATIVE_STATUS_3_OFFSET,
+
+    LAPC_SNAPSHOT_STATUS_0_OFFSET, LAPC_SNAPSHOT_STATUS_1_OFFSET,
+    LAPC_SNAPSHOT_STATUS_2_OFFSET, LAPC_SNAPSHOT_STATUS_3_OFFSET
+  };
+
+  uint32_t currData[XLAPC_STATUS_PER_SLOT];
+  
+  for (int c=0; c < XLAPC_STATUS_PER_SLOT; c++) {
+#ifndef _WIN32
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+    size += xclRead(handle, XCL_ADDR_SPACE_DEVICE_CHECKER, dbgIpInfo->m_base_address+statusRegisters[c], &currData[c], sizeof(uint32_t));
+#ifndef _WIN32
+#pragma GCC diagnostic pop
+#endif
+  }
+
+  lapcResults.OverallStatus[index]      = currData[XLAPC_OVERALL_STATUS];
+  std::copy(currData+XLAPC_CUMULATIVE_STATUS_0, currData+XLAPC_SNAPSHOT_STATUS_0, lapcResults.CumulativeStatus[index]);
+  std::copy(currData+XLAPC_SNAPSHOT_STATUS_0, currData+XLAPC_STATUS_PER_SLOT, lapcResults.SnapshotStatus[index]);
+
+  (void)size;
+
+  // increment debugIpNum
+  ++debugIpNum[LAPC];
+  lapcResults.NumSlots = (unsigned int)debugIpNum[LAPC];
+}
+
+void 
+DebugIpStatusCollector::printLAPCResults(std::ostream& _output)
+{ 
+  if(0 == lapcResults.NumSlots) {
+    return;
+  }
+
+  bool violations_found = false;
+  bool invalid_codes = false;
+  _output << "\nLight Weight AXI Protocol Checkers codes \n";
+  auto col1 = std::max(cuNameMaxStrLen[LAPC], strlen("CU Name")) + 4;
+  auto col2 = std::max(portNameMaxStrLen[LAPC], strlen("AXI Portname"));
+
+  for (size_t i = 0; i < lapcResults.NumSlots; ++i) {
+    if (!xclAXICheckerCodes::isValidAXICheckerCodes(lapcResults.OverallStatus[i],
+                lapcResults.SnapshotStatus[i], lapcResults.CumulativeStatus[i])) {
+      _output << "CU Name: " << cuNames[LAPC][i] << " AXI Port: " << portNames[LAPC][i] << "\n";
```

Comment:
Please use boost::format.

Reply:
I added use of boost::format while printing the LAPC violations. At this current line, there is no formatting used. So, I am not adding any change here.